src/hidimstat/base_perturbation.py

```diff
@@ -42,6 +42,14 @@ def __init__(
             The method used for making predictions. This determines the predictions
             passed to the loss function. Supported methods are "predict",
             "predict_proba", "decision_function", "transform".
+        feature_groups: dict or None, default=None
+            A dictionary where the keys are the group names and the values are the
+            list of column names corresponding to each features group. If None,
+            the feature_groups are identified based on the columns of X.
+        feature_types: str or list, default="auto"
```

Comment:
Should feature_types be an attribute of `BasePerturbation`? It would only be used for CFI. If we want to add it here, we should also add the `categorical_max_cardinality` in the same class.

Reply:
I think that makes more sense to avoid running it in methods for which the data type does not matter (e.g. LOCO).